docs/source/generics.rst

```diff
@@ -335,6 +328,43 @@ possibly by making use of the ``Any`` type.
 
 For some advanced uses of self-types see :ref:`additional examples <advanced_self>`.
 
+Automatic self types using typing.Self
+**************************************
+
+The patterns described above are quite common, so there is a syntactic sugar
+for them introduced in :pep:`673`. Instead of defining a type variable and
+using an explicit ``self`` annotation, you can import a magic type ``typing.Self``
+that is automatically transformed into a type variable with an upper bound of
+current class, and you don't need an annotation for ``self`` (or ``cls`` for
+class methods). The above example can thus be rewritten as:
+
+.. code-block:: python
+
+   from typing import Self
+
+   class Friend:
+       other: Self | None = None
+
+       @classmethod
+       def make_pair(cls) -> tuple[Self, Self]:
+           a, b = cls(), cls()
+           a.other = b
+           b.other = a
+           return a, b
+
+   class SuperFriend(Friend):
+       pass
+
+   a, b = SuperFriend.make_pair()
+
+This is more compact than using explicit type variables, plus additionally
+you can use ``Self`` in attribute annotations, not just in methods.
+
+.. note::
+
+   This feature is available on Python 3.11 or newer. On older Python versions
+   you can import backported ``Self`` from latest ``typing_extensions``.
```

Comment:
I'd suggest replacing 'latest' with 'recent enough' since otherwise it go stale quickly.

Reply:
Also, it's misleading to start with "This feature is available on Python 3.11 or newer." Users will think they can't use `Self` before 3.11. I'd suggest saying:
```suggestion
   ``Self`` can be imported from ``typing`` on Python 3.11 or higher, and also from
   ``typing_extensions`` on version 4.0.0 or higher.
```
(Ref: https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#release-400-november-14-2021)

Reply:
Yes, I will make it more clear (or maybe just commit what Jelle proposed)